packages/clay-link/src/index.tsx

```diff
@@ -85,14 +131,28 @@ const ClayLink = React.forwardRef<HTMLAnchorElement, IProps>(
 				[`btn-${displayType}`]: displayType && !outline && !borderless,
 				[`btn-outline-${displayType}`]:
 					displayType && (outline || borderless),
+				[weight ? FONT_WEIGHTS[weight] : '']: weight,
+				[`text-${fontSize}`]: fontSize,
+				[`text-decoration-${decoration}`]: decoration,
 			};
 		} else {
+			if (displayType && LINK_PRESETS[displayType]) {
+				if (decoration !== null && !outline) {
+					decoration =
+						decoration || LINK_PRESETS[displayType]?.decoration;
+				}
+			}
+
 			classes = {
 				'link-monospaced': monospaced,
 				'link-outline': outline,
 				'link-outline-borderless': borderless,
-				[`link-${displayType}`]: displayType && !outline,
+				[displayType ? LINK_PRESETS[displayType]?.base : '']:
+					displayType && !outline,
 				[`link-outline-${displayType}`]: displayType && outline,
+				[weight ? FONT_WEIGHTS[weight] : '']: weight,
```

Comment:
I think the ternary would not be needed here because it will already only render if `weight` is set.

```suggestion
				[FONT_WEIGHTS[weight!]]: weight,
```

Reply:
Thanks for this, I was wondering how to get around typescript's complaints.